### PR TITLE
chore: Remove ModuleTemplate from releases

### PR DESCRIPTION
**Pull request by shorim**
## Description

Changes proposed in this pull request (what was done and why):

- Stop creating ModuleTemplate in Github releases
- Remove verify-module Github action
- Remove documentation for installation using ModuleTemplate
- Add a new Github action `Github Release` which waits for the `post-telemetry-manager-build-release` job to succeed and then executes the make release target
- Update the documentation for the release process.

Changes refer to particular issues, PRs or documents:

- #566 

## Traceability
- [x] The PR is linked to a GitHub issue.
- [x] New features have a milestone set.
- [x] New features have defined acceptance criteria in a corresponding GitHub Issue, and all criteria are satisfied with this PR.
- [x] The corresponding GitHub issue has a respective `area` and `kind` label.
- [ ] The follow-up issues (if any) are linked in the `Related Issues` section.
- [ ] Adjusted the documentation if the change is user-facing.
- [ ] The feature is unit-tested
- [ ] The feature is e2e-tested

### CODE REVIEW

File: .github/workflows/push-release.yml [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+      - '[0-9]+.[0-9]+.[0-9]+'
+jobs:
+  release:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout repo
+        uses: actions/checkout@v4
+
+      # wait for the build to succeed so that the manager image is available
+      - name: Wait for the 'push-telemetry-manager-build' job to succeed
```
Comment (hisarbalik) [inline .github/workflows/push-release.yml:15]: The name shouldn't be somethings like `Wait for the 'post-telemetry-manager-build-release' job to succeed` ?

Comment (shorim) [reply .github/workflows/push-release.yml]: Good catch! yes, you are correct

**Review by hisarbalik — DISMISSED** [review]

File: docs/contributor/releasing.md [hunk 2]
```diff
@@ -45,11 +43,11 @@ This release process covers the steps to release new major and minor versions fo
    git push upstream {RELEASE_VERSION}
    git push upstream {RELEASE_DEV_VERSION}
    ```
-   The {RELEASE_VERSION} tag triggers a postsubmit job (`post-telemetry-manager-release-module`) that creates the GitHub release.
+   The {RELEASE_VERSION} tag triggers a post-submit prow job (`post-telemetry-manager-build-release`) and a Github action (`Github Release`). The `post-telemetry-manager-build-release` job builds the `telemetry-manager` image, tags it with the module version and pushes it to the production registry. The `Github Release` action creates the Github release.
```
Comment (IwonaLanger) [inline docs/contributor/releasing.md:46]:
```suggestion
   The {RELEASE_VERSION} tag triggers a post-submit Prow Job (`post-telemetry-manager-build-release`) and a GitHub action (`Github Release`). The `post-telemetry-manager-build-release` job builds the `telemetry-manager` image, tags it with the module version, and pushes it to the production registry. The `GitHub Release` action creates the GitHub release.
```

File: docs/contributor/releasing.md [hunk 3]
```diff
@@ -45,11 +43,11 @@ This release process covers the steps to release new major and minor versions fo
...
 
-9. Verify the [Prow Status](https://status.build.kyma-project.io/) of the postsubmit job (`post-telemetry-manager-release-module`).
-   - Once the postsubmit job succeeds, the new Github release is available under [releases](https://github.com/kyma-project/telemetry-manager/releases).
-   - If the postsubmit job fails, retrigger it by removing the {RELEASE_VERSION} tag from upstream and pushing it again:
+9. Verify the [status](https://status.build.kyma-project.io/) of the post-submit prow job (`post-telemetry-manager-build-release`) and the [status](https://github.com/kyma-project/telemetry-manager/actions) of the Github action (`Github Release`).
```
Comment (IwonaLanger) [inline docs/contributor/releasing.md:48]:
```suggestion
9. Verify the [status](https://status.build.kyma-project.io/) of the post-submit Prow Job (`post-telemetry-manager-build-release`) and the [status](https://github.com/kyma-project/telemetry-manager/actions) of the GitHub action (`Github Release`).
```

File: docs/contributor/releasing.md [hunk 4]
```diff
@@ -45,11 +43,11 @@ This release process covers the steps to release new major and minor versions fo
...
+   - Once the post-submit prow job and the Github action succeed, the new Github release is available under [releases](https://github.com/kyma-project/telemetry-manager/releases).
```
Comment (IwonaLanger) [inline docs/contributor/releasing.md:49]:
```suggestion
   - Once the post-submit Prow Job and the GitHub action succeed, the new GitHub release is available under [releases](https://github.com/kyma-project/telemetry-manager/releases).
```

File: docs/contributor/releasing.md [hunk 5]
```diff
@@ -45,11 +43,11 @@ This release process covers the steps to release new major and minor versions fo
...
+   - If the post-submit prow job or the Github action fails, re-trigger them by removing the {RELEASE_VERSION} tag from upstream and pushing it again:
```
Comment (IwonaLanger) [inline docs/contributor/releasing.md:50]:
```suggestion
   - If the post-submit Prow Job or the GitHub action fails, re-trigger them by removing the {RELEASE_VERSION} tag from upstream and pushing it again:
```

**Review by hisarbalik — APPROVED** [review]

**Pull request merged**